web-components/src/components/news-feed/news-feed.ts

```diff
@@ -0,0 +1,548 @@
+        endDate.setUTCHours(23, 59, 59, 999)
+        if (now > endDate) {
+          return false // Already ended
+        }
+      }
+
+      // 3. Check language (using original 'languages' array if present on item)
+      if (item.languages && item.languages.length > 0) {
+        const langMatch = item.languages.some(
+          (lang) =>
+            currentLang.toLowerCase() === lang.toLowerCase() ||
+            currentLang.startsWith(lang.toLowerCase() + "-")
+        )
+        if (!langMatch) {
+          return false
+        }
+      }
+
+      // 4. Check country (using original 'countries' array if present on item)
+      if (currentCountry && item.countries && item.countries.length > 0) {
+        const countryMatch = item.countries.some(
+          (country) => currentCountry.toLowerCase() === country.toLowerCase()
+        )
+        if (!countryMatch) {
+          return false
+        }
+      }
+
+      // 5. Check App Version (using min/max versions from item)
+      if (currentAppVersion) {
+        const versionCompare = (v1: string | undefined, v2: string | undefined) => {
+          const parts1 = (v1 || "").split(/[.-]/)
+          const parts2 = (v2 || "").split(/[.-]/)
+          for (let i = 0; i < Math.max(parts1.length, parts2.length); i++) {
+            const p1Str = parts1[i] || "0"
+            const p2Str = parts2[i] || "0"
+            const p1Num = parseInt(p1Str, 10)
+            const p2Num = parseInt(p2Str, 10)
+            if (!isNaN(p1Num) && !isNaN(p2Num)) {
+              if (p1Num > p2Num) return 1
+              if (p1Num < p2Num) return -1
+            } else {
+              if (p1Str > p2Str) return 1
+              if (p1Str < p2Str) return -1
+            }
+          }
+          return 0
+        }
+
+        if (item.min_app_version && versionCompare(currentAppVersion, item.min_app_version) < 0) {
+          return false
+        }
+        if (item.max_app_version && versionCompare(currentAppVersion, item.max_app_version) > 0) {
+          return false
+        }
+      }
+
+      // If all checks pass
+      return true
+    })
+  }
+
+  // --- Lit Render Methods ---
+
+  _renderNewsItem(item: ProcessedNewsItem) {
+    const hasUrl = item.url && item.url.trim() !== ""
+    const placeholderId = `placeholder-${item.id || Math.random().toString(36).substring(7)}`
+
+    // Create placeholder icon content
+    const placeholderContent = item.id ? item.id.substring(0, 3) : "NWS"
+
+    // Create the icon html - either image with fallback or just placeholder
+    const iconTemplate = item.icon_url
+      ? html`
+          <img
+            src=${item.icon_url}
+            alt=""
+            class="icon"
+            @error=${(e: Event) => {
+              ;(e.target as HTMLImageElement).style.display = "none"
+              const element = this.shadowRoot!.querySelector(`#${placeholderId}`)! as HTMLElement
+              element.style.display = "flex"
+            }}
+          />
```

Comment:
Do we really have this icon thing ?  (where are those icons files).
If it's the case, document a bit (but I fear it's kind of an hallucination).